### PR TITLE
Improved GPT4All model streaming tokens without adding callbacks

### PR DESCRIPTION
**Pull request by Anindyadeep**
- **Description:** Right now token streaming is enabled by adding callbacks. But there was no streaming of tokens occuring after adding callbacks.Latest version of gpt4all: version `1.0.3` addresses this issue and hence I tried to integrate that in the existing function.I have kept the code for supporting callbacks for streaming,  however I think that can be removed in coming versions. Also added docs for the same.
  
  - **Issue:** This commit fixes issue #7747.,
  - Dependencies: No dependency change,
  - Tag maintainer: @hwchase17 @agola11,
  - Twitter handle: @AnindyadeepS

### CODE REVIEW

**Comment by Anindyadeep** [comment]
@hwchase17 @baskaryan Can you please review this PR and help me to understand why the lint checks for python 3.8 failed?

File: langchain/llms/gpt4all.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 """Wrapper for the GPT4All model."""
+import io
 from functools import partial
 from typing import Any, Dict, List, Mapping, Optional, Set
 
+import pkg_resources
```
Comment (hinthornw) [inline langchain/llms/gpt4all.py:6]: I think we should opt for a different stdlib package if possible https://discuss.python.org/t/will-setuptools-remove-pkg-resource-module-in-the-future/27182

Comment (Anindyadeep) [reply langchain/llms/gpt4all.py:6]: @hinthornw We can do it in another way, the streaming for gpt4all throws `NotImplementedError` if used the previous version, hence we can detect there by a `try-except` block. In that way we can do it without `pkg_resource` 

Another thing, should we remove the callback support for streaming and make this changes as the default one? In that way there might be less ambiguity?

Comment (Anindyadeep) [reply langchain/llms/gpt4all.py:6]: @hinthornw I also wonder, Does providing custom StreamingCallback provide any kind of added advantage?

**Comment by Anindyadeep** [comment]
@hinthornw can you please review this and provide your thoughts
Thanks

**Comment by Anindyadeep** [comment]
@hinthornw , I am closing this PR, as I see now that adding this support explicitely might break the overall flow of how langchain's LLM wrapper works. As every wrapper is following a convention, along with how it will stream. So should I close this PR?

**Pull request closed without merging**